File: sorting/bubbleSort.js
const swap = (array, i, j) => {
  const temp = array[i];
  array[i] = array[j];
  array[j] = temp;
};
// Time complexity: O(n^2)
// Space complexity: O(1)
const bubbleSort = (array) => {
  for (let i = 0; i < array.length; i++) {
    for (let j = 0; j <= i; j++) {
      if (array[i] < array[j]) {
        swap(array, i, j);
      }
    }
  }
  return array;
};

File: sorting/bubbleSort.py
from typing import List


def bubbleSort(nums: List[int]) -> List[int]:
  n = len(nums)
  for i in range(n):
    for j in range(i):
      if nums[i] < nums[j]:
        nums[i], nums[j] = nums[j], nums[i]
  return nums
# Time complexity: O(n^2)
# Space complexity: O(1)